app/models/concerns/type_of_means_assessment.rb

```diff
@@ -77,6 +77,18 @@ def benefit_check_recipient
 
   private
 
+  # involvement_in_case is stored on partner_detail when a database applications and
+  # partner when a datastore application.
+  def partner_involvement_in_case
+    partner_detail&.involvement_in_case || partner&.involvement_in_case
+  end
+
+  # conflict_of_interest is stored on partner_detail when a database applications and
+  # partner when a datastore application.
+  def partner_conflict_of_interest
+    partner_detail&.conflict_of_interest || partner&.conflict_of_interest
+  end
+
```

Comment:
Required to allow this method to work with both datastore and database applications. Ideally this would not be necessary.